src/SurveyService.Data/GroupRepository.cs

```diff
@@ -13,6 +18,34 @@ public class GroupRepository : IGroupRepository
   public GroupRepository(IDataProvider provider)
   {
     _provider = provider;
+  }
+
+  private IQueryable<DbGroup> CreateGetPredicates(
+      GetGroupFilter filter,
+      IQueryable<DbGroup> query)
+  {
+    if (filter.IncludeQuestions)

```

Comment:
I'd reduce nesting by changing conditions like that:
`if (filter.IncludeQuestions)
    {
      query = query.Include(group => group.Questions.Where(question => question.IsActive));
    }

    if (filter.IncludeQuestions && filter.IncludeOptions)
    {
      query = query
        .Include(group => group.Questions.Where(question => question.IsActive))
        .ThenInclude(question => question.Options.Where(option => option.IsActive && !option.IsCustom));
    }

    if (filter.IncludeQuestions && filter.IncludeOptions && filter.IncludeUserAnswers)
    {
      query = query
        .Include(group => group.Questions.Where(question => question.IsActive))
        .ThenInclude(question => question.Options.Where(option => option.IsActive))
        .ThenInclude(option => option.UsersAnswers);
    }

    return query;`